src/OrchardCore.Modules/OrchardCore.Settings/Services/SetupEventHandler.cs

```diff
@@ -17,23 +18,15 @@ public SetupEventHandler(ISiteService siteService)
         }
 
         public async Task Setup(
-            string siteName,
-            string userName,
-            string userId,
-            string email,
-            string password,
-            string dbProvider,
-            string dbConnectionString,
-            string dbTablePrefix,
-            string siteTimeZone,
+            IDictionary<string, object> properties,
             Action<string, string> reportError
             )
         {
             // Updating site settings
             var siteSettings = await _siteService.LoadSiteSettingsAsync();
-            siteSettings.SiteName = siteName;
-            siteSettings.SuperUser = userId;
-            siteSettings.TimeZoneId = siteTimeZone;
+            siteSettings.SiteName = properties["SiteName"]?.ToString();
```

Comment:
Can we define these as `static readonly` fields in a class, so we can avoid magic string issues that may occurs

Reply:
done

Reply:
If so, why not just using `nameof(SetupContext.SiteName)`, in place of creating a class for this

Note: There are many other places where we use "magic" strings, but that may be okay when they are quite obvious.

Reply:
> nameof(SetupContext.SiteName)

I was thinking about that, nothing but there are few cases like `siteTimeZone` not equivalent to `TimeZoneId`

Reply:
So here it would just be

    siteSettings.TimeZoneId = properties[nameof(SetupContext.SiteTimeZone)]?.ToString()

In place of

    siteSettings.TimeZoneId = properties[SetupConstants.SiteTimeZone]?.ToString()

Here the advantage of the nameof is that if you change a setup context prop name, it will not compile.

Reply:
Both are fine, for 

```csharp
siteSettings.TimeZoneId = properties[SetupConstants.SiteTimeZone]?.ToString()
```

if the field has been changed, the value still the same ;) I never mind to use anyone of them instead of magic strings

Reply:
SetupContext has been stripped down and SiteName property and all other string properties have been removed and palced in the Properties bag.